### PR TITLE
Recommend MailPoet in free users emails - MAILPOET-3981

### PR DESCRIPTION
**Pull request by triple0t**
[MAILPOET-3981](https://mailpoet.atlassian.net/browse/MAILPOET-3981)

### CODE REVIEW

File: lib/Config/ServicesChecker.php [hunk 1]
```diff
@@ -132,6 +132,23 @@ public function isMailPoetAPIKeyPendingApproval(): bool {
     return $mssActive && $mssKeyValid && $mssKeyPendingApproval;
   }
 
+  public function isUserActivelyPaying(): bool {
+    $isPremiumKeyValid = $this->isPremiumKeyValid(false);
+    $premiumSupportTier = $this->settings->get('premium.premium_key_state.data.support_tier');
+    $isUserPayingForPremium = $premiumSupportTier === 'premium';
```
Comment (wxa) [inline lib/Config/ServicesChecker.php:138]: There is already a method for checking this: `$this->subscribersFeature->hasPremiumSupport()`. It's worth keeping such logic in one place by reusing the existing method.

I know that `ServicesChecker` and `SubscribersFeature` have some duplicate checks for legacy reasons. Originally `ServicesChecker` was used for handling license-related error notices, and `SubscribersFeature` for checking if the subscriber limit is reached, but over time their functionality started to overlap as both needed more granular checks for various license-related cases. Ripe for refactoring, it seems.

File: lib/Config/ServicesChecker.php [hunk 2]
```diff
@@ -132,6 +132,23 @@ public function isMailPoetAPIKeyPendingApproval(): bool {
...
+
+    $mssActive = Bridge::isMPSendingServiceEnabled();
+    $isMssKeyValid = $this->isMailPoetAPIKeyValid(false);
+    $mssSupportTier = $this->settings->get('mta.mailpoet_api_key_state.data.support_tier');
+    $isUserPayingForMss = $mssSupportTier === 'premium';
```
Comment (wxa) [inline lib/Config/ServicesChecker.php:143]: Same as above: maybe it's better to create a method similar to `hasPremiumSupport()` in `MailPoet\Util\License\Features\Subscribers`.

**Review by wxa — CHANGES_REQUESTED** [review]

File: lib/Config/ServicesChecker.php (hunk 1 above)
Comment (triple0t) [reply lib/Config/ServicesChecker.php:138]: Noted.
I wasn’t aware of that method
Thanks for the review @wxa  🙌🏿

**Review by wxa — DISMISSED** [review]

**Review by veljkho — APPROVED** [review]

**Review by veljkho — APPROVED** [review]

**Comment by triple0t** [comment]
Closing due to Git Merge errors
New PR here: https://github.com/mailpoet/mailpoet/pull/3863

**Pull request closed without merging**